builder/xenserver/iso/step_create_instance.go

```diff
@@ -141,35 +141,39 @@ func (self *stepCreateInstance) Run(state multistep.StateBag) multistep.StepActi
 			return multistep.ActionHalt
 		}
 
-	} else {
-		// Look up the network by it's name label
-
-		networks, err := client.GetNetworkByNameLabel(config.NetworkName)
+		_, err = instance.ConnectNetwork(network, "0")
 
 		if err != nil {
-			ui.Error(fmt.Sprintf("Error occured getting Network by name-label: %s", err.Error()))
-			return multistep.ActionHalt
+			ui.Say(err.Error())
 		}
 
-		switch {
-		case len(networks) == 0:
-			ui.Error(fmt.Sprintf("Couldn't find a network with the specified name-label '%s'. Aborting.", config.NetworkName))
-			return multistep.ActionHalt
-		case len(networks) > 1:
-			ui.Error(fmt.Sprintf("Found more than one network with the name '%s'. The name must be unique. Aborting.", config.NetworkName))
-			return multistep.ActionHalt
-		}
+	} else {
+		// Look up each network by it's name label
+		for vif, networkNameLabel := range config.NetworkName {
```

Comment:
Given 'vif' here is just a index - it would be clearer to label as such (or something like i, or x).